impl/src/main/java/org/kuali/rice/kim/web/struts/action/IdentityManagementGroupDocumentAction.java

```diff
@@ -228,6 +229,14 @@ public ActionForward addMember(ActionMapping mapping, ActionForm form, HttpServl
         }
         if(checkKimDocumentGroupMember(newMember) && 
         		KRADServiceLocatorWeb.getKualiRuleService().applyRules(new AddGroupMemberEvent("", groupDocumentForm.getGroupDocument(), newMember))){
+
+			// Set member's fullname if possible
+			Entity entity = KimApiServiceLocator.getIdentityService().getEntityByPrincipalName(newMember.getMemberName());

```

Comment:
It is possible, and I get the sense that this code is less than optimal as originally written -- though, where do we draw the line for refactoring? I could see weird edge cases creeping through here, and so introduced the least amount of risk by not changing this house of cards. (honestly, look at the rest of the class, it is horrible)